oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -453,7 +453,7 @@ public Builder jwkSetUri(String jwkSetUri) {
 		 * @return the {@link Builder}
 		 */
 		public Builder providerConfigurationMetadata(Map<String, Object> configurationMetadata) {
-			if (!CollectionUtils.isEmpty(configurationMetadata)) {
+			if (configurationMetadata != null) {
```

Comment:
Can we add a test since this was a problem?

Reply:
@rwinch I added a couple of tests